testing/upload_validation.ts

```diff
@@ -1202,8 +1202,34 @@ const unrefinedPackVersionMetadataSchema = zodCompleteObject<PackVersionMetadata
     message: 'Formula namespaces can only contain alphanumeric characters and underscores.',
   }),
   systemConnectionAuthentication: z.union(zodUnionInput(systemAuthenticationValidators)).optional(),
-  formulas: z.array(formulaMetadataSchema).max(Limits.BuildingBlockCountPerType).optional().default([]),
-  formats: z.array(formatMetadataSchema).max(Limits.BuildingBlockCountPerType).optional().default([]),
+  formulas: z
+    .array(formulaMetadataSchema)
+    .max(Limits.BuildingBlockCountPerType)
+    .optional()
+    .default([])
+    .superRefine((data, context) => {
+      const formulaNames = data.map(formulaDef => formulaDef.name);
+      for (const dupe of getNonUniqueElements(formulaNames)) {
```

Comment:
Ah this is my bad, but can we make getNonUniqueElements() case-insensitive? The formula language will treat formulas the same regardless of case, so we should do the same here.

Reply:
Had to do a questionable bit of type casting to make it case insensitive. Would be worth checking if that's what you had in mind